src/GraphQL/Language/AST/SourceLocation.cs

```diff
@@ -3,63 +3,32 @@ namespace GraphQL.Language.AST
     /// <summary>
     /// Provides information regarding the location of a node within a document's original query text.
     /// </summary>
-    public class SourceLocation
+    public readonly struct SourceLocation
     {
         /// <summary>
         /// Initializes a new instance with the specified parameters.
         /// </summary>
-        public SourceLocation(int line, int column, int start = -1, int end = -1)
+        public SourceLocation(int start = -1, int end = -1)
         {
-            Line = line;
-            Column = column;
-            Start = start;
-            End = end;
+            _start = start + 1;
+            _end = end + 1;
```

Comment:
This allows for identifying an empty struct.  It should be equally as fast as adding a bool to the struct while using less memory.